### DISCUSSION TITLE
Clustered installation on machines not perfectly time-synchronized

### DISCUSSION THREAD

**Discussion by 5292874** [quartznet/quartznet#2248] [General]
Hello everybody, 

I would like to ask if anybody knows what would be the problems that I could encounter if I installed Quartz.NET on several clustered machines which are not perfectly time-synchronized. The Quartz.NET instances would be pointing to the same SQL database of course.
 
Could the data in the database get corrupted?
 
Or maybe the same jobs could be triggered twice or more times? If this was the _only_ problem then I was thinking about putting in place an external mechanism so that the job knows that it has already run. When that happens the job could then simply just exit without doing anything.
 
Basically I am trying to see if there is a workaround in case the time synchronization cannot be guaranteed.
 
Any help or suggestion would be highly appreciated.

**Comment by risperdal** [comment]
I’m facing the same question. In a clustered Quartz.NET setup, what really happens if the nodes are off by a few seconds?

Can this lead to job duplication or misfires?

Or is the database consistency guaranteed by Quartz itself?

If duplication is the only risk, is the recommended approach to implement idempotency checks inside the job?
